tasks/quality_gates.py

```diff
@@ -0,0 +1,141 @@
+            body_info += f"|{SUCCESS_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            with_info = True
+        else:
+            body_error += f"|{FAIL_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            body_error_footer += f"|{gate['name']}|{gate['error_type']}|{gate['message'].replace("\n", "<br>")}|\n"
+            with_error = True
+
+    body_error_footer += "\n</details>\n"
+    body = f"Please find below the results from static quality gates\n{body_error+body_error_footer if with_error else ''}\n\n{body_info if with_info else ''}"
+
+    pr_commenter(ctx, title=title, body=body)
+
+
+def _print_quality_gates_report(gate_states: list[dict[str, typing.Any]]):
+    print(color_message("======== Static Quality Gates Report ========", "magenta"))
+    for gate in sorted(gate_states, key=lambda x: x["error_type"] is not None):
+        if gate["error_type"] is None:
+            print(color_message(f"Gate {gate['name']} succeeded {SUCCESS_CHAR}", "blue"))
+        elif gate["error_type"] == "AssertionError":
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} because of the following assertion failures :\n{gate['message']}",
+                    "orange",
+                )
+            )
+        else:
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} with the following stack trace :\n{gate['message']}",
+                    "orange",
+                )
+            )
+
+
+@task
+def parse_and_trigger_gates(ctx, config_path="test/static/static_quality_gates.yml"):
+    """
+    Parse and executes static quality gates
+    :param ctx: Invoke context
+    :param config_path: Static quality gates configuration file path
+    :return:
+    """
+    with open(config_path) as file:
+        config = yaml.safe_load(file)
+
+    gate_list = list(config.keys())
+    quality_gates_mod = __import__("tasks.static_quality_gates", fromlist=gate_list)
+    print(f"{config_path} correctly parsed !")
+    metric_handler = GateMetricHandler(
+        git_ref=os.environ["CI_COMMIT_REF_SLUG"], bucket_branch=os.environ["BUCKET_BRANCH"]
+    )
+
+    print(f"The following gates are going to run:\n\t- {"\n\t- ".join(gate_list)}")
+    final_state = "success"
+    gate_states = []
+    for gate in gate_list:
+        gate_inputs = config[gate]
+        gate_inputs["ctx"] = ctx
+        gate_inputs["metricHandler"] = metric_handler
+        try:
+            gate_mod = getattr(quality_gates_mod, gate)
+            gate_mod.entrypoint(**gate_inputs)
+            print(f"Gate {gate} succeeded !")
+            gate_states.append({"name": gate, "state": True, "error_type": None, "message": None})
```

Comment:
Could you walk me through the thought process behind using a dictionary here instead of alternatives (e.g., a dataclass or a class)? Are you expecting a flexible data schema for gate states?

Reply:
Resolving in favor of https://github.com/DataDog/datadog-agent/pull/32964#issuecomment-2625254367.